lib/src/language.c

```diff
@@ -24,6 +24,29 @@ uint32_t ts_language_state_count(const TSLanguage *self) {
   return self->state_count;
 }
 
+uint32_t ts_language_supertype_count(const TSLanguage *self) {
+  return self->version >= LANGUAGE_VERSION_WITH_SUPERTYPE_INFO ? self->supertype_count : 0;
```

Comment:
Gate this with `LANGUAGE_VERSION_WITH_RESERVED_WORDS` instead, and remove the other define